City-of-Helsinki/drupal-hdbt
# UHF-2074 adjanced accordions without margin

[UHF-2074](https://helsinkisolutionoffice.atlassian.net/browse/UHF-2074)

## What was done

* Library component wrapping div was removed to make handling components easier.
* Removed top margin from all accordions that follow these rules:
  * Accordion comes after another accordion of the same style (white or grey)
  * The latter accordion has no title and no description entered

## What was out of scope for this ticket:

* Accordions between upper and lower regions still get margin between them.
* Accordions that are hardcoded (like accessibility information on some pages) still have margin to possible surrounding accordions.

In this screenshot, all accordions contain two accordion items, some of them are moved adjacent according to rules above. 
![image](https://user-images.githubusercontent.com/1191667/138654161-bf1af99f-f142-4aaa-afb7-afb327c8b197.png)

## How to test

* Get branch, `drush cr`
* Create a [Basic page](https://helfi.docker.so/en/node/add/page)
* Add accordions to page.
  * Add at least two adjancent accordions with same style and latter one without main accordion title or description
  * Add also accordions with different styles and titles and descriptions
* Check that accordions follow the rules above.
